src/cart/src/gurt/hash.c

```diff
@@ -690,58 +788,105 @@ d_hash_table_create_inplace(uint32_t feats, unsigned int bits, void *priv,
 	htable->ht_ops	 = hops;
 	htable->ht_priv	 = priv;
 
-	D_ALLOC_ARRAY(buckets, nr);
-	if (buckets == NULL)
-		return -DER_NOMEM;
+	if (hops->hop_rec_hash == NULL && !(feats & D_HASH_FT_NOLOCK)) {
+		htable->ht_feats |= D_HASH_FT_GLOCK;
+		D_WARN("The d_hash_table_ops_t->hop_rec_hash() callback is "
```

Comment:
Ah, so it's not really optional.  This restriction should be placed in the header file.

Reply:
Put comment in header file.